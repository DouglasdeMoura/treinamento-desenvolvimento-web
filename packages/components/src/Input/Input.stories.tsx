import React from 'react';
import type { Meta } from '@storybook/react';

import { Input } from './Input';

export default {
  title: 'Input',
  component: Input,
} as Meta;

const Template = (args) => <Input {...args} />;

export const Primary = Template.bind({});

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
  value: 'Input value',
};