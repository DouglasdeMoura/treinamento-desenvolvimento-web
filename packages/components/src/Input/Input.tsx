import React, { forwardRef } from 'react';
import styled from 'styled-components';

type InputProps = {
  color?: string,
  background?: string,
} & JSX.IntrinsicElements['input'];

const InputComponent = forwardRef<HTMLButtonElement, InputProps>(({
  ...rest
}, ref) => (
  <input {...rest} ref={ref} />
));

export const Input = styled(InputComponent)`
  border: 2px solid var(--color-gray-400);
  border-radius: 4px;
  color: ${props => props.color ?? 'var(--color-gray-600)'};
  cursor: not-allowed;
  background: ${props => props.background ?? 'white'};
  padding: 0.5em 1em;

  :disabled {
    background: var(--color-gray-100);
    border-color: var(--color-gray-300);
    color: var(--color-gray-300);
  }

  &:focus {
    border-color: var(--color-gray-700);
  }
`;
