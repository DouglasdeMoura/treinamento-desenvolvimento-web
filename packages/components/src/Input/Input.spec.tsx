import { Input } from './Input';
import renderer from 'react-test-renderer';

describe('<Input />', () => {
  it('deve renderizar o Input', () => {
    const tree = renderer
      .create(<Input />)
      .toJSON();
    
      expect(tree).toMatchSnapshot();
  });
});
