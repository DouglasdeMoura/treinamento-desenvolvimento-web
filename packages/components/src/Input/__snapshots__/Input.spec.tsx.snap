// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> deve renderizar o Input 1`] = `
<input
  className="sc-bdfBQB dSzMHI"
/>
`;
